modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryRelatedObjectsResourceImpl.java

```diff
@@ -42,13 +45,46 @@
 	public ObjectEntryRelatedObjectsResourceImpl(
 		ObjectDefinitionLocalService objectDefinitionLocalService,
 		ObjectEntryManagerRegistry objectEntryManagerRegistry,
+		ObjectRelatedModelsProviderRegistry objectRelatedModelsProviderRegistry,
 		ObjectRelationshipService objectRelationshipService) {
 
 		_objectDefinitionLocalService = objectDefinitionLocalService;
 		_objectEntryManagerRegistry = objectEntryManagerRegistry;
+		_objectRelatedModelsProviderRegistry =
+			objectRelatedModelsProviderRegistry;
 		_objectRelationshipService = objectRelationshipService;
 	}
 
+	@Override
+	public void deleteCurrentObjectEntry(
+			Long currentObjectEntryId, String objectRelationshipName,
+			Long relatedObjectEntryId)
+		throws Exception {
+
+		ObjectEntryManager objectEntryManager =
+			_objectEntryManagerRegistry.getObjectEntryManager(
+				_objectDefinition.getStorageType());
+
+		_checkCurrentObjectEntry(objectEntryManager, currentObjectEntryId);
+
+		_checkRelatedObjectEntry(
+			objectEntryManager, objectRelationshipName, relatedObjectEntryId);
+
+		ObjectRelationship objectRelationship =
+			_objectRelationshipService.getObjectRelationship(
+				_objectDefinition.getObjectDefinitionId(),
+				objectRelationshipName);
+
+		ObjectRelatedModelsProvider objectRelatedModelsProvider =
+			_objectRelatedModelsProviderRegistry.getObjectRelatedModelsProvider(
+				_objectDefinition.getClassName(), objectRelationship.getType());
+
+		objectRelatedModelsProvider.disassociateRelatedModels(
+			GuestOrUserUtil.getUserId(),
```

Comment:
Can we use the `contextUser` parameter instead?

Reply:
Makes sense! I think that is better to use `context` instead of `GuestOrUserUtil.getUserId()`